### DISCUSSION TITLE
Plotting a function with X/Y Plot widget

### DISCUSSION THREAD

**Discussion by mirek23** [ControlSystemStudio/phoebus#2616] [Q&A]
Hello,

How to plot the function with supplied parameters trough PVs using the X/Y Plot.
let's assume that I have two PVs: PV_A and PV_B and I want to draw a function:
f(x)=PV_A*x+PV_B
I tried to use the scripts using the traces[0].y_value but the script did not work.
It is probably the  traces[0].y_value does not accept indexing like  traces[0].y_value[i] ?
or maybe it is another way?

Best Regards
Mirek

**Comment by georgweiss** [comment]
Would it be possible to calculate f(x) in the IOC and expose that as a "derived" PV?

Another way to solve it is to use one of the built-in formula functions. Using these is much more efficient than executing scripts. For your use case: if x values are represented in an array PV named x_values, you could for the Y axis use the "scale" formula function like so:

```=scale(x_values, PV_A, PV_B)```

**Comment by mirek23** [comment]
Hi Georg,

In order to do that what you have suggested I need the X Y widget Property PV X to be set. 
Is there a possibility to setup or generate the local waveform (within Phoebus screen) in the range for example [0,100] to be used for X Y widget Property PV X?
Where could I find a documentation or examples with the scale function and other functions of this kind?

Best Regards
Mirek

**Comment by georgweiss** [comment]
Array formula functions are documented in the Help. Search for formula and select topic "Formula" (not "Formula Functions").

There are additional formula functions besides array manipulation, but these are "documented in code", see for instance package https://github.com/ControlSystemStudio/phoebus/tree/master/core/formula/src/main/java/org/csstudio/apputil/formula
